app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VotesController < ApplicationController
+  before_action :find_micropost
+  
+  def create
+
+    if already_voted?
+      flash[:warning] = "You can't vote more than once"
+    else
+      # I know it is not safely because someone can hack database by coding in form (or somehow).
+      # Does validation in model (1..5) solve this problem ?
+     @vote = @micropost.votes.create(grades: params[:grades], user_id: current_user.id) 
+      if @vote.save
+        redirect_to root_url
+      else
+        flash[:warning] = "1, 2, 3, 4 or 5 only"
+        redirect_to root_url
+      end
+    end
+      
+  end
+
+    private
+
+  # I wrote this before I created the thing that hides the form. 
+  # Let keep it here yet
```

Comment:
That's a really bad practice. You have no reason to keep code that's not in use.
If you reeeeeally have to keep some code because you think it will be of use - make a commit with it and then remove it in a separate commit.

In a perfect world, any repo has only working code.